dash/development/_py_prop_typing.py

```diff
@@ -0,0 +1,77 @@
+def generate_any(_):
+    return "typing.Any"
+
+
+def generate_shape(t):
+    props = []
+
+    for prop in t["value"].values():
+        prop_type = PROP_TYPING.get(prop["name"], generate_any)(prop)
+        if prop_type not in props:
+            props.append(prop_type)
+
+    if len(props) == 0:
+        return "typing.Any"
+
+    return f"typing.Dict[str, typing.Union[{', '.join(props)}]]"
+
+
+def generate_union(t):
+    types = []
+    for union in t["value"]:
+        u_type = PROP_TYPING.get(union["name"], generate_any)(union)
+        if u_type not in types:
+            types.append(u_type)
+    return f"typing.Union[{', '.join(types)}]"
+
+
+def generate_tuple(type_info):
+    els = type_info.get("elements")
+    elements = ", ".join(get_prop_typing(x.get("name"), x) for x in els)
+    return f"typing.Tuple[{elements}]"
+
+
+def generate_array_of(t):
+    typed = PROP_TYPING.get(t["value"]["name"], generate_any)(t["value"])
+    return f"typing.List[{typed}]"
+
+
+def generate_object_of(t):
```

Comment:
Object can have any hashable type as keys, not just strs.

Reply:
> Object can have any hashable type as keys

They're all going to turn into strings when we JSON-encode them though. Sometimes people use numbers as the keys, which has some important use cases so I guess we had better not prevent it, but it also causes problems, eg does `5.0` encode to `'5'` or `'5.0'`? https://github.com/plotly/dash/issues/1760